packages/dataviews/src/view-list.tsx

```diff
@@ -28,24 +39,41 @@ import type {
 	ViewList as ViewListType,
 } from './types';
 
+import { ActionsDropdownMenuGroup, ActionModal } from './item-actions';
+
+interface Action {
+	callback: ( items: Item[] ) => void;
+	icon: any;
```

Comment:
I'm not sure how to provide the type for this. Happy to prepare a follow-up to improve it.

Reply:
I've actually provided types for all the actions in #61400 but unfortunately my PR is blocked because of some unrelated failure. Maybe we should just extract the action types from my PR and land it (without actually typing item-actions.js). I'll try this in its own PR and we can reuse these types here instead.

Reply:
I actually tried adding `IconType` as you do in your PR, but that made this PR fail locally.